src/psyclone/tests/psyir/transformations/block_loop_trans_test.py

```diff
@@ -0,0 +1,360 @@
+    blocktrans.apply(schedule.children[0])
+    code = str(psy.gen)
+    print(code)
+    correct = \
+        '''DO j_out_var = cu_fld%internal%ystart, cu_fld%internal%ystop, 32
+        j_el_inner = MIN(j_out_var + 32, cu_fld%internal%ystop)
+        DO j = j_out_var, j_el_inner, 1
+          DO i = cu_fld%internal%xstart, cu_fld%internal%xstop, 1
+    '''
+    assert correct in code
+    correct = '''END DO
+        END DO
+      END DO'''
+    assert correct in code
+
+
+def test_blockloop_trans_apply_neg():
+    '''Test the apply function of BlockLoopTrans for a negative step'''
+    _, invoke_info = parse(os.path.join(GOCEAN_BASE_PATH, "single_invoke.f90"),
+                           api="gocean1.0")
+    psy = PSyFactory("gocean1.0", distributed_memory=False).\
+        create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    schedule.children[0].children[2].replace_with(Literal("-1", INTEGER_TYPE))
+    blocktrans = BlockLoopTrans()
+    blocktrans.apply(schedule.children[0])
+    code = str(psy.gen)
+    correct = \
+        '''DO j_out_var = cu_fld%internal%ystart, cu_fld%internal%ystop, -32
+        j_el_inner = MAX(j_out_var - 32, cu_fld%internal%ystop)
+        DO j = j_out_var, j_el_inner, -1
+          DO i = cu_fld%internal%xstart, cu_fld%internal%xstop, 1
+    '''
+    assert correct in code
+    correct = '''END DO
+        END DO
+      END DO'''
+    assert correct in code
+
+
+def test_blockloop_trans_apply_double_block():
+    '''Test the apply function of BlockLoopTrans for multiple
+    blocks of 2 nested loops'''
+    code = \
+        '''Program test
+    integer :: i, j, end
+    integer, dimension(1:end, 1:end) :: ai, aj
+    do i=1, end
+        do j=1, end
+            ai(i, j) = 1
+        end do
+    end do
+
+    do i=1, end, 2
+      do j = 1, end, 2
+        aj(i, j) = 1
+      end do
+    end do
+    End Program test'''
+    reader = FortranReader()
+    psyir = reader.psyir_from_source(code)
+    blocktrans = BlockLoopTrans()
+    loops = psyir.walk(Loop)
+    for loop in loops:
+        blocktrans.apply(loop)
+    writer = FortranWriter()
+    result = writer(psyir)
+    correct_vars = \
+        '''integer :: i_el_inner
+  integer :: i_out_var
+  integer :: j_el_inner
+  integer :: j_out_var'''
+    assert correct_vars in result
+
+    correct = \
+        '''do i_out_var = 1, end, 32
+    i_el_inner = MIN(i_out_var + 32, end)
+    do i = i_out_var, i_el_inner, 1
+      do j_out_var = 1, end, 32
+        j_el_inner = MIN(j_out_var + 32, end)
+        do j = j_out_var, j_el_inner, 1
+          ai(i,j) = 1
+        enddo
+      enddo
+    enddo
+  enddo
+  do i_out_var = 1, end, 32
+    i_el_inner = MIN(i_out_var + 32, end)
+    do i = i_out_var, i_el_inner, 2
+      do j_out_var = 1, end, 32
+        j_el_inner = MIN(j_out_var + 32, end)
+        do j = j_out_var, j_el_inner, 2
+          aj(i,j) = 1
+        enddo
+      enddo
+    enddo
+  enddo'''
+    assert correct in result
```

Comment:
Can you add:
(at the top)
from psyclone.tests.utilities import Compile

(at the test signature)
def test_blockloop_trans_apply_double_block(tmpdir):

(and here)
assert Compile(tmpdir).string_compiles(result)

To make sure the generated Fortran is valid (you may need the "end" integer to be a parameter and other small fixes)

Reply:
Done, it still just compiles when I run the test though.

Reply:
Can you bring the "assert Compile(tmpdir).string_compiles(result)" statement to the end. I think it makes sense we first check the produced code is as expected and then we try to compile.

I don't understand your response but to test it you need to do:
`pytest --compile src/psyclone/tests/psyir/transformations/block_loop_trans_test.py`
and at the moment this gives a compiler error

Reply:
Sorry I missed this. It now does work with `pytest --compile`